core/jvm/src/test/scala/scalaz/zio/ZManagedSpec.scala

```diff
@@ -283,37 +276,46 @@ class ZManagedSpec(implicit ee: org.specs2.concurrent.ExecutionEnv) extends Test
     (retries1 must be_===(3)) && (retries2 must be_===(3))
   }
 
-  private def zipParFailAcquisition = {
-    val first  = ZManaged.unit
-    val second = ZManaged.reserve(Reservation(ZIO.fail(()), ZIO.unit))
-
-    unsafeRunSync(first.zipPar(second).use(_ => ZIO.unit)) must be_===(Exit.Failure(Cause.Fail(())))
-  }
-
-  private def zipParFailReservation = {
-    val first  = ZManaged.unit
-    val second = ZManaged.fromEffect(ZIO.fail(()))
-
-    unsafeRunSync(first.zipPar(second).use(_ => ZIO.unit)) must be_===(Exit.Failure(Cause.Fail(())))
-  }
+  private def zipParFailAcquisition =
+    unsafeRunSync {
+      for {
+        latch  <- Promise.make[Nothing, Unit]
+        first  = ZManaged.fromEffect(latch.succeed(()) *> ZIO.sleep(Duration.Infinity))
+        second = ZManaged.reserve(Reservation(latch.await *> ZIO.fail(()), ZIO.unit))
+        _      <- first.zipPar(second).use_(ZIO.unit)
+      } yield ()
+    } must be_===(Exit.Failure(Cause.Both(Cause.Fail(()), Cause.Interrupt)))
+
+  private def zipParFailReservation =
+    unsafeRunSync {
+      for {
+        latch  <- Promise.make[Nothing, Unit]
+        first  = ZManaged.fromEffect(latch.succeed(()) *> ZIO.sleep(Duration.Infinity))
+        second = ZManaged.fromEffect(latch.await *> ZIO.fail(()))
+        _      <- first.zipPar(second).use_(ZIO.unit)
+      } yield ()
+    } must be_===(Exit.Failure(Cause.Both(Cause.Fail(()), Cause.Interrupt)))
 
   private def zipParFailAcquisitionRelease = {
     var releases = 0
     val first    = ZManaged.unit
     val second   = ZManaged.reserve(Reservation(ZIO.fail(()), ZIO.effectTotal(releases += 1)))
 
-    val exit = unsafeRunSync(first.zipPar(second).use(_ => ZIO.unit))
-    (exit must be_===(Exit.Failure(Cause.Fail(())))) && (releases must be_===(1))
+    unsafeRunSync(first.zipPar(second).use(_ => ZIO.unit))
+    releases must be_===(1)
   }
 
-  private def zipParFailReservationRelease = {
-    var releases = 0
-    val first    = ZManaged.reserve(Reservation(ZIO.unit, ZIO.effectTotal(releases += 1)))
-    val second   = ZManaged.make(ZIO.fail(()))(_ => ZIO.effectTotal(releases += 1))
-
-    val exit = unsafeRunSync(first.zipPar(second).use(_ => ZIO.unit))
-    (exit must be_===(Exit.Failure(Cause.Fail(())))) && (releases must be_===(1))
-  }
+  private def zipParFailReservationRelease =
+    unsafeRun {
+      for {
+        reserveLatch <- Promise.make[Nothing, Unit]
```

Comment:
There is a `withLatch` combinator courtesy of @regiskuckaertz that we may want to take a look at.

Reply:
Just took a look at it.
Looks awesome and we should definitely use it :+1: 

Question is whether is should update this pr or make another one were everything is migrated over to withLatch? I would prefer the latter option as we would immediately profit from reduced flakiness :grin:

Reply:
Then let's do the latter!

Reply:
Cool :+1: 
What needs to be still changed in this pr?